### PR TITLE
Fix constness of array.d

### PR DESCRIPTION
**Pull request by nordlow**
Thanks to @aG0aep6G this works as far as I can see.

### CODE REVIEW

**Comment by nordlow** [comment]
Is there something you expected to do here?

**Comment by nordlow** [comment]
Ping.

**Comment by quickfur** [comment]
ping @DmitryOlshansky @JakobOvrum @MartinNowak @andralex

File: std/container/array.d [hunk 1]
```diff
@@ -2,9 +2,161 @@ module std.container.array;
 
 import std.range.primitives;
 import std.traits;
+import core.exception : RangeError;
+import std.algorithm : move;
 
 public import std.container.util;
 
+private static struct RangeT(A)
```
Comment (JakobOvrum) [inline std/container/array.d]: `static` for structs is a no-op at module level

**Comment by JakobOvrum** [comment]
I'd still prefer a more general solution involving conversions between `const(C.Range!T)` and `C.Range!(const T)`, but this patch seems forward-compatible with such a solution (except possible deprecation of `C.ConstRange` and `C.ImmutableRange`), so LGTM.

**Comment by JakobOvrum** [comment]
I'll merge it in a day or so unless someone finds new evidence against it until then.

File: std/container/array.d (hunk 1 above)
Comment (nordlow) [inline std/container/array.d]: Shall I remove?

Comment (JakobOvrum) [inline std/container/array.d]: Yes please :)

Comment (nordlow) [inline std/container/array.d]: Done.

**Comment by JakobOvrum** [comment]
Quite a few commits here, please squash.

**Comment by andralex** [comment]
squash

**Comment by andralex** [comment]
after that green light @JakobOvrum

**Comment by nordlow** [comment]
Done.

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Pull request merged**